### ISSUE TITLE
Steal Swallows Errors When Parsing JSON

### ISSUE THREAD

**Issue by joe-crick** [stealjs/steal#763]
When parsing JSON files, Steal swallows errors, as below:

```js
function parse(load){
		if ( (load.metadata.format === 'json' || !load.metadata.format) && jsonTest.test(load.source)  ) {
			try {
				return JSON.parse(load.source);
			} catch(e) {}  // <-- Swallows error  
		}

	}
```
This can result in JSON files failing to load when imported, with no indication of why.

**Comment by matthewp** [comment]
We should probably warn. Think the reason why it swallows the errors is because we don't know for sure that the user intends this to parsed as json and we are assuming so via the regular expression.  So I think a warning here is the way to go.

**Comment by matthewp** [comment]
Fixed by #884
